fl4health/clients/nnunet_client.py

```diff
@@ -577,7 +642,16 @@ def compute_loss_and_additional_losses(
                 f"Got {len(loss_preds)} predictions and {len(loss_targets)} targets."
             )
 
-        return self.criterion(loss_preds, loss_targets), None
+        # If device type is cuda, nnUNet defaults to compute loss in mixed precision
+        # https://github.com/MIC-DKFZ/nnUNet/blob/43349fa5f0680e8109a78dca7215c19e258c9dd7 \
+        # nnunetv2/training/nnUNetTrainer/nnUNetTrainer.py#L993
+        if self.device.type == "cuda":
+            with torch.autocast(self.device.type, enabled=True):
+                loss = self.criterion(loss_preds, loss_targets), None
```

Comment:
Out of curiosity, does autocast automatically cast variables back to the default precision when the scope is exited? For example are the loss values here going to remain fp16 or whatever they were cast to?

Reply:
Yeah I think you are right!